src/script/interpreter.cpp

```diff
@@ -1681,6 +1750,218 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
+                                stack.push_back(vchOutNonce);
+                            }
+                            break;
+                        }
+                        case OP_INSPECTOUTPUTSCRIPTPUBKEY:
+                        {
+                            const PrecomputedTransactionData *cache = checker.GetPrecomputedTransactionData();
+                            // Return error if the evaluation context is unavailable
+                            // TODO: Handle accoding to MissingDataBehavior
+                            if (!cache)
+                                return set_error(serror, SCRIPT_ERR_INTROSPECT_CONTEXT_UNAVAILABLE);
+                            valtype vchScriptPubKeySha256;
+                            vchScriptPubKeySha256.insert(vchScriptPubKeySha256.begin(), cache->m_output_spk_single_hashes[idx].begin(), cache->m_output_spk_single_hashes[idx].begin() + 32);
+                            pushspk(stack, out.scriptPubKey, vchScriptPubKeySha256);
+                            break;
+                        }
+                        default: assert(!"invalid opcode"); break;
+                    }
+                }
+                break;
+
+                case OP_INSPECTVERSION:
+                case OP_INSPECTLOCKTIME:
+                case OP_INSPECTNUMINPUTS:
+                case OP_INSPECTNUMOUTPUTS:
+                case OP_TXWEIGHT:
+                {
+                    // Transaction introspection is available post tapscript
+                    if (sigversion == SigVersion::BASE || sigversion == SigVersion::WITNESS_V0) return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
+
+                    switch (opcode)
+                    {
+                        case OP_INSPECTVERSION:
+                        {
+                            push4_le(stack, static_cast<uint32_t>(checker.GetTxVersion()));
+                            break;
+                        }
+                        case OP_INSPECTLOCKTIME:
+                        {
+                            push4_le(stack, checker.GetLockTime());
+                            break;
+                        }
+                        case OP_INSPECTNUMINPUTS:
+                        {
+                            uint64_t num_ins = checker.GetTxvIn()->size();
+                            assert(num_ins <= MAX_SIZE);
+                            stack.push_back(CScriptNum(static_cast<int64_t>(num_ins)).getvch());
+                            break;
+                        }
+                        case OP_INSPECTNUMOUTPUTS:
+                        {
+                            // No need to bound checks for num_outputs. Elements consensus rules would break
+                            // if outpoint more 2**30 is allowed.
+                            uint64_t num_outs = checker.GetTxvOut()->size();
```

Comment:
You have to check to see if `checker.GetTxvOut()` is null and return false in that case, with a TODO to assert failure based on the `mdb` value.

Reply:
This type should be valtype::size_type I believe.